test/integration/acceptance/query_permission_test_ast_txs.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TEST_INTEGRATION_ACCEPTANCE_QUERY_PERMISSION_TEST_AST_TXS_HPP_H_
+#define IROHA_TEST_INTEGRATION_ACCEPTANCE_QUERY_PERMISSION_TEST_AST_TXS_HPP_H_
+
+#include "query_permission_test_base.hpp"
+
+using namespace shared_model;
+using namespace integration_framework;
+using namespace shared_model::interface::permissions;
+
+class QueryPermissionAssetTxs final : public QueryPermissionTestBase {
+ public:
+
+  QueryPermissionAssetTxs();
+
+  /**
+   * Prepare state of ledger:
+   * - execute transfer asset from admin to target account
+   * - execute transfer asset from target to admin account
+   *
+   * @param spectator_permissions - set of query permisisons for target user's
+   * and spectators' accounts
+   * @return reference to ITF
+   */
+  IntegrationTestFramework &prepareState(
+      AcceptanceFixture *fixture,
```

Comment:
Why via the pointer and not reference? Also seems missed in a doc

Reply:
The reference is not used because derived classes with overriding members are passed there.

Reply:
not a problem, actually https://ideone.com/oBqrZ4

Reply:
you are right. changed to references.